Makefile

```diff
@@ -1,12 +1,36 @@
-# Copyright 2019 Amazon.com, Inc. or its affiliates. All Rights Reserved.
+# Copyright 2019-2021 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 # SPDX-License-Identifier: Apache-2.0
 
-BUILDDIR ?= $(abspath build)
-OUTDIR   ?= $(abspath blobs)
+BUILDDIR                      ?= $(abspath build)
+OUTDIR                        ?= $(abspath blobs)
+ARCH_x86_64                    = x86_64
+ARCH_aarch64                   = aarch64
+LINUX_aarch64                  = arm64
+KERN_IMAGE_x86_64              = bzImage
+KERN_IMAGE_aarch64             = Image
+CONFIG_TEMPLATE                = configs/microvm-kernel-config-
+DEFAULT_AARCH64_CROSS_COMPILER = /usr/bin/aarch64-linux-gnu-
+HOST_MACHINE                   = $(shell uname -m)
+BUILD_ARCH                    ?= $(HOST_MACHINE)
+
+CONFIG=$(CONFIG_TEMPLATE)$(BUILD_ARCH)
+
+ifeq ($(BUILD_ARCH),$(ARCH_x86_64))
+	KERN_IMAGE=$(KERN_IMAGE_x86_64)
+	ARCH=$(ARCH_x86_64)
+else ifeq ($(BUILD_ARCH),$(ARCH_aarch64))
+	KERN_IMAGE=$(KERN_IMAGE_aarch64)
+	ifndef CROSS_COMPILE
+		ifneq ($(ARCH),$(HOST_MACHINE))
```

Comment:
Shouldn't here be BUILD_ARCH instead of ARCH?

Reply:
yes it should, thanks for the notice